packages/govuk-frontend/src/govuk/errors/index.mjs

```diff
@@ -59,7 +52,7 @@ export class ElementError extends GOVUKFrontendError {
    * @param {object} options - Element error options
    * @param {string} options.componentName - The name of the component throwing the error
    * @param {string} options.identifier - An identifier that'll let the user understand which element has an error (variable name, CSS selector)
-   * @param {typeof HTMLElement} [options.expectedType] - The type that was expected for the element
+   * @param {typeof HTMLElement | StringConstructor} [options.expectedType] - The type that was expected for the identifier
```

Comment:
Hope this makes sense

We have errors on the **Skip link** `$module` element [`.hash` property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement/hash) which now output:

* `'Skip link: $module.hash is not an instance of "String"'`
* `'Skip link: $module.hash target #this-element-does-not-exist not found'`

Reply:
Is `StringConstructor` purely a TypeScript thing? I can't find vanilla JS docs for it.

And the second question: do we care? I haven't been on top of all the parameter typing, but have we included things elsewhere that aren't either native or explicitly imported?

To be clear: I think this is fine and understandable, just curious about these queries.

Reply:
Yeah `StringConstructor` is in their library types with `BooleanConstructor`, `NumberConstructor` etc

https://github.com/microsoft/TypeScript/blob/main/src/lib/es5.d.ts#L526
https://github.com/Microsoft/TypeScript/blob/main/src/lib/es2015.core.d.ts#L522

I'm happy with them since we we heavily rely on their DOM libs already

Context wise, I made the switch from `typeof String` to `StringConstructor` to stop ESLint fighting:

> `Invalid JSDoc @param "options.expectedType" type "String"; prefer: "string".`
> eslint([jsdoc/check-types](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/check-types.md#repos-sticky-header))

But this morning I'm having second thoughts since it's a primitive `'string'` not a `new String('string')`

Reply:
Should probably make `expectedType` a string since it's only for building a message

Reply:
@domoscargin How's this? https://github.com/alphagov/govuk-frontend/compare/f825dbacee1080a6088c165fee749b609c6a359f..70265f15802a4a9f599e4f7b71e15632d10d62ee

Reply:
Looks good to me! Tests will probably need an update for the slightly different messages.